MapboxSearch/sdk/src/main/java/com/mapbox/search/details/RetrieveDetailsOptions.kt

```diff
@@ -0,0 +1,96 @@
+package com.mapbox.search.details
+
+import android.os.Parcelable
+import com.mapbox.search.AttributeSet
+import com.mapbox.search.base.ExperimentalMapboxSearchAPI
+import com.mapbox.search.base.defaultLocaleLanguage
+import com.mapbox.search.common.IsoCountryCode
+import com.mapbox.search.common.IsoLanguageCode
+import com.mapbox.search.internal.bindgen.DetailsOptions
+import com.mapbox.search.mapToCore
+import kotlinx.parcelize.Parcelize
+
+/**
+ * Options, used for the [DetailsApi.retrieveDetails].
+ */
+@ExperimentalMapboxSearchAPI
+@Parcelize
+public class RetrieveDetailsOptions @JvmOverloads constructor(
+
+    /**
+     * Besides the basic metadata attributes, developers can request additional
+     * attributes by setting attribute_sets parameter with attribute set values,
+     * for example &attribute_sets=basic,photos,visit.
+     * The requested metadata will be provided in metadata object in the response.
+     */
+    public val attributeSets: List<AttributeSet>? = null,
+
+    /**
+     * Specify the user’s language. This parameter controls the language of the text supplied in responses.
+     * If language is not set explicitly, then language from default system locale will be used.
+     */
+    public val language: IsoLanguageCode = defaultLocaleLanguage(),
+
+    /**
+     * The ISO country code to requests a worldview for the location data,
+     * if applicable data is available.
+     * This parameters will only be applicable for Boundaries and Places feature types.
+     */
+    public val worldview: IsoCountryCode? = null,
+) : Parcelable {
+
+    /**
+     * @suppress
+     */
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (javaClass != other?.javaClass) return false
+
+        other as RetrieveDetailsOptions
+
+        if (attributeSets != other.attributeSets) return false
+        if (language != other.language) return false
+        if (worldview != other.worldview) return false
+
+        return true
+    }
+
+    /**
+     * @suppress
+     */
+    override fun hashCode(): Int {
+        var result = attributeSets?.hashCode() ?: 0
+        result = 31 * result + language.hashCode()
+        result = 31 * result + (worldview?.hashCode() ?: 0)
+        return result
+    }
```

Comment:
nit: could just be
```
override fun hashCode(): Int = listOf(attributeSets.orEmpty(), language, worldview).hashCode()
```

Reply:
It's autogenerated by the IDE, I'd left it as is.